alfi/driver.py

```diff
@@ -113,13 +117,26 @@ def run_solver(solver, res, args):
             with DumbCheckpoint(chkptdir + "nssolution-Re-%s" % (re), mode=FILE_READ) as checkpoint:
                 checkpoint.load(solver.z, name="up_%i" % re)
         except:
-            (z, info_dict) = solver.solve(re)
-            results[re] = info_dict
-            if args.checkpoint:
-                with DumbCheckpoint(chkptdir + "nssolution-Re-%s" % (re), mode=FILE_UPDATE) as checkpoint:
-                    checkpoint.store(solver.z, name="up_%i" % re)
+            try:
+                if from_zero_each_time:
+                    solver.z.assign(0)
+                (z, info_dict) = solver.solve(re)
+                results[re] = info_dict
+                if args.checkpoint:
+                    with DumbCheckpoint(chkptdir + "nssolution-Re-%s" % (re), mode=FILE_UPDATE) as checkpoint:
+                        checkpoint.store(solver.z, name="up_%i" % re)
+            except:
+                results[re] = {
+                    "Re": re,
+                    "nu": solver.nu.values()[0],
+                    "linear_iter": float('nan'),
+                    "nonlinear_iter": float('nan'),
+                    "time": float('nan'),
+                }
+                break
+
         if args.paraview:
-            pvdf.write(solver.visprolong(solver.z.split()[0]), time=re)
+            pvdf.write(*solver.z.split(), time=re)
```

Comment:
This is no longer needed because Paraview does high order I suppose.

Reply:
Paraview's high-order support isn't great. In particular I havent figured out how to coax paraview into drawing isosurfaces using high-order information. I'm still saving refined PVD files for degree 3 for other problems ..